tests/datetime.test/t6_01.req

```diff
@@ -0,0 +1 @@
+select (cast(cast(now() as text) as datetime) - cast(cast(now() as text) as datetime)) as intv
```

Comment:
In this case, would sqlite compute NOW() only once? If not, the result could be non-zero.

Reply:
Actually it does. It stores NOW() in a register and reuses it.

Reply:
Yep. The bug seems to be hit only when we cast `now()` to text first and cast the text back to datetime again. I use the expression (essentially `now() = now()`) so I can always get the same output.